vendor/engines/split_accounts/app/services/split_accounts/order_detail_list_transformer.rb

```diff
@@ -0,0 +1,25 @@
+module SplitAccounts
+
+  class OrderDetailListTransformer
+
+    attr_reader :order_details
+
+    def initialize(order_details = [])
+      @order_details = order_details
+    end
+
+    def perform
+      order_details.map do |order_detail|
```

Comment:
I also tried `order_details.flat_map` and while that's an improvement over the `map`/`flatten`, it still tended to be slower than the `each_with_object`, running around ~5.2 seconds.